Source/Core/DolphinQt/Debugger/MemoryViewWidget.cpp

```diff
@@ -295,29 +341,53 @@ void MemoryViewWidget::Update()
       });
       break;
     }
-
-    if (row_breakpoint)
-    {
-      bp_item->setData(Qt::DecorationRole, Resources::GetScaledThemeIcon("debugger_breakpoint")
-                                               .pixmap(QSize(rowHeight(0) - 3, rowHeight(0) - 3)));
-    }
   }
+}
 
-  setColumnWidth(0, rowHeight(0));
+void MemoryViewWidget::UpdateBreakpointTags()
+{
+  if (Core::GetState() != Core::State::Paused)
+    return;
 
-  // Number of characters possible.
-  int max_length = GetCharacterCount(m_type);
+  for (int i = 0; i < rowCount(); i++)
+  {
+    bool row_breakpoint = true;
+
+    for (int c = 2; c < columnCount(); c++)
+    {
+      // Pull address from cell itself, helpful for dual column view.
+      auto cell = item(i, c);
+      u32 address = cell->data(Qt::UserRole).toUInt();
 
-  // Column width is the max number of characters + 1 or 2 for the space between columns. A longer
-  // length means less columns, so a bigger spacing is fine.
-  max_length += max_length < 8 ? 1 : 2;
-  const int width = m_font_width * max_length;
+      if (address == 0)
+      {
+        row_breakpoint = false;
+        continue;
+      }
 
-  for (int i = 2; i < columnCount(); i++)
-    setColumnWidth(i, width);
+      // In dual view the only sizes that dont match up on both left and right views are for
+      // Double, which uses two or four columns of hex32.
+      int type_size = GetTypeSize(m_type);
+      if (m_dual_view && m_type == Type::Double && c < (columnCount() == 8 ? 6 : 4))
```

Comment:
This condition on the right there is magic number nonsense. There's gotta be a better way to express this.